pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -499,6 +508,11 @@ func (p *PodAutoscalerInternal) BuildStatus(currentTime metav1.Time, currentStat
 	}
 	status.Conditions = append(status.Conditions, newCondition(rolloutStatus, verticalReason, currentTime, datadoghq.DatadogPodAutoscalerVerticalAbleToApply, existingConditions))
 
+	for _, condition := range status.Conditions {
+		if condition.Status == corev1.ConditionTrue {
+			autoscalingStatusConditions.Inc(string(condition.Type), condition.Reason, condition.Message)
```

Comment:
It's probably more a 0-1 gauge (1 = True, 0 = everything else). The telemetry should probably be in the controller code and not in the `model` package.